online/src/app/products-list/products-list.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { ProductViewModel } from '../products';
+import { AuthentificationService } from '../services/authentification.service';
+import { ProductService } from '../services/product.service';
+import { admin, customer } from '../utils';
+
+@Component({
+  selector: 'app-products-list',
+  templateUrl: './products-list.component.html',
+  styleUrls: ['./products-list.component.scss']
+})
+export class ProductsListComponent implements OnInit {
+  productID: number = -1;
+  products: ProductViewModel[] | undefined;
+  hasCustomerRole = this.authService.hasRoleType(customer);
+  hasAdminRole = this.authService.hasRoleType(admin);
+  
+  constructor(private productService: ProductService, private authService: AuthentificationService) { }
+
+
+  ngOnInit(): void {
+    this.getProductsList();
+  }
+
+  getProductsList(): void {
+    this.productService.getProducts().subscribe(products => this.products = products);
+  }
+
+  addToCartHandler(): void {
+    this.productService.addToCart(this.productID);
+  }
+
+  refreshID(id: number) {
```

Comment:
What does this do?

Reply:
I needed to be aware of the product ID the customer wanted to add to cart, so that I could display the details in the shopping cart component, hence that function was just a handler for doing so.